kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```
@@ -233,6 +233,8 @@ protected TransactionImpl buildTransactionInternal(
     latestSnapshot.ifPresent(
         snapshot -> validateWriteToExistingTable(engine, snapshot, isCreateOrReplace));
     validateTransactionInputs(engine, isCreateOrReplace);
+    checkArgument(
+        isCreateOrReplace || !schema.isPresent(), "Schema evolution is not supported currently");
```

Of course if someone just removes the blanket ignore in the schema evolution tests, we'll see this failure but @allisonport-db let me know if you'd prefer an explicit test for this. It didn't seem worth it since it's just scoped for 4.0 and I didn't want to deviate too much from what exists in master.

---

I think I'd prefer just a very simple test to make sure we are indeed blocking it -- I see some tests failing, maybe fixing those to intercept this error can be sufficient?

---

Also minor but can we make this error message a little more specific? ie. schema evolution is unsupported -- a schema can only be provided for a new table

---

Should we throw `UnsupportedOperationException`?

---

+1, a simple unit test would be great